### PR TITLE
dovecot: update to 2.2.28 and add SQLite support

### PR DESCRIPTION
**Pull request by lucize**
disable icu #4077 as is not yet upstream
also add me as the maintainer

split #4093 
Compile/Run tested x86_64 LEDE/Reboot SNAPSHOT r3640-f229f4a
Signed-off-by: Lucian Cristian lucian.cristian@gmail.com

### CODE REVIEW

**Comment by stintel** [comment]
Please split the first commit into 3 separate commits. This would also solve the problem of the too long commit subject.

**Comment by lucize** [comment]
splitted

regards

File: mail/dovecot/Makefile [hunk 1]
```diff
@@ -65,6 +71,14 @@ else
...
 endif
 
+ifneq ($(strip $(CONFIG_DOVECOT_SQLITE)),)
+  CONFIGURE_ARGS+= \
+        --with-sqlite=yes
+else
+  CONFIGURE_ARGS+= \
+        --with-sqlite=no
+endif
+
```
Comment (stintel) [inline mail/dovecot/Makefile]: Can be handled in CONFIGURE_ARGS directly, see https://git.lede-project.org/?p=source.git;a=blob;f=package/network/services/lldpd/Makefile;h=de7de03cc86af175f945bd9ad96e4b5637390e67;hb=HEAD#l101

**Comment by lucize** [comment]
done

regards

File: mail/dovecot/Makefile [hunk 2]
```diff
@@ -56,15 +61,9 @@ CONFIGURE_ARGS += \
...
-
-ifneq ($(strip $(CONFIG_DOVECOT_LDAP)),)
-  CONFIGURE_ARGS+= \
-	--with-ldap=yes
-else
-  CONFIGURE_ARGS+= \
-	--with-ldap=no
-endif
+	--with-icu=no \
+$(if $(CONFIG_DOVECOT_LDAP),--with-ldap=yes,--with-ldap=no) \
```
Comment (stintel) [inline mail/dovecot/Makefile]: Please also move the LDAP change to a separate commit. Preferably before this commit. Thanks.

**Comment by lucize** [comment]
splitted again

**Comment by stintel** [comment]
@tripolar LGTM, please merge if you agree with maintainership transfer.

**Comment by lucize** [comment]
@stintel he agreed in #4093

**Comment by nxhack** [comment]
@lucize 

Oops. Unfortunately we miss '+'.
```
-  DEPENDS:=+DOVECOT_LDAP:libopenldap DOVECOT_SQLITE:libsqlite3 +libopenssl +librt +zlib +libbz2 +libcap
+  DEPENDS:=+DOVECOT_LDAP:libopenldap +DOVECOT_SQLITE:libsqlite3 +libopenssl +librt +zlib +libbz2 +libcap
```

**Comment by lucize** [comment]
sorry about that, missed it after splitting it
